### ISSUE TITLE
[10.0] Feature Request - small

### ISSUE THREAD

**Issue by gdgellatly** [odoo/odoo#17960]
Impacted versions:
9.0, 10.0 - maybe earlier but we had web_context_tunnel for this.
Steps to reproduce:
Create a view and xpath inheritance for attributes for a field with context (product_id of sale_order form is a good one)
```
<xpath expr="//field[@name='order_line']/tree//field[@name='product_id']" position="attributes">
                    <attribute name="context" add="'{'attribute_value_id': parent.attribute_value_id}'" remove="company_id" />
```

Current behavior:

It nearly works - apply_inheritance_specs does everything you expect, except there is no way to remove the trailing } from the original value (not with above string but mangling to try and make work).

Expected behavior:

That add and remove work for all attributes.  Just check if the attribute is a context, domains, or attrs and if it is treat it slightly differently so that add and remove can function here rather than ugly workarounds with fields_view_get.

The spec I'd expect is in the case of attrs and context
add - string dict
remove - key

domains
add and remove expression tuples.

I'm only running into this in context, but can see same issue in attrs and domain.

**Comment by pedrobaeza** [comment]
Implemented in #26607 :wink:

**Comment by pedrobaeza** [comment]
I'm going to close this issue in favor of the opened PR, although that one is still debatable.
